code/05_further_clean_other_data_and_compute_item_missingness.R

```diff
@@ -0,0 +1,376 @@
+                                  fromLast = TRUE), ]
+
+# ---------------------------------------------------------------------------- #
+# Recode "prefer not to answer" values ----
+# ---------------------------------------------------------------------------- #
+
+# Check for substantive analysis variables. None already have NA values.
+
+sum(is.na(dat3$oa[, oa_items])) == 0
+
+sum(is.na(dat3$anxiety_identity$anxiety_identity)) == 0
+sum(is.na(dat3$mechanisms$comp_act_willing)) == 0
+sum(is.na(dat3$mental_health_history$anxiety_duration)) == 0
+
+sum(dat3$oa[, oa_items] == 555, na.rm = TRUE)
+
+sum(dat3$anxiety_identity$anxiety_identity == 555, na.rm = TRUE)
+sum(dat3$mechanisms$comp_act_willing == 555, na.rm = TRUE)
+sum(dat3$mental_health_history$anxiety_duration == 555, na.rm = TRUE)
+
+dat3$oa[, oa_items][dat3$oa[, oa_items] == 555] <- NA
+
+dat3$anxiety_identity$anxiety_identity[dat3$anxiety_identity$anxiety_identity == 555] <- NA
+dat3$mechanisms$comp_act_willing[dat3$mechanisms$comp_act_willing == 555] <- NA
+dat3$mental_health_history$anxiety_duration[dat3$mental_health_history$anxiety_duration == 555] <- NA
+
+# Check for DASS-21-AS. "dass21_as" already has NA values. At "Eligibility", 555 
+# ("prefer not to answer") was already recoded to NA for "_mean" items, and original 
+# items were recoded as NA. For "dass21_as" at other time points, "_mean" items are 
+# NA (as they were computed only at "Eligibility").
+
+sum(is.na(dat3$dass21_as[dat3$dass21_as$session_only != "Eligibility", dass21_as_items]))
+sum(is.na(dat3$dass21_as[dat3$dass21_as$session_only == "Eligibility", dass21_as_mean_items]))
+
+sum(dat3$dass21_as[, dass21_as_items] == 555, na.rm = TRUE)
+sum(dat3$dass21_as[, dass21_as_mean_items] == 555, na.rm = TRUE)
+
+dat3$dass21_as[, dass21_as_items][dat3$dass21_as[, dass21_as_items] == 555] <- NA
+
+# ---------------------------------------------------------------------------- #
+# Check response ranges ----
+# ---------------------------------------------------------------------------- #
+
+# Check for substantive analysis variables
+
+all(sort(unique(as.vector(as.matrix(dat3$oa[, oa_items])))) %in% 0:4)
+
+all(sort(unique(dat3$anxiety_identity$anxiety_identity)) %in% 0:4)
+all(sort(unique(dat3$mechanisms$comp_act_willing)) %in% 1:7)
+all(sort(unique(dat3$mental_health_history$anxiety_duration)) %in% 0:8)
+
+# Check for DASS-21-AS
+
+all(sort(unique(as.vector(as.matrix(dat3$dass21_as[, dass21_as_items])))) %in% 0:3)
+
+dass21_as_mean_range <- 
+  range(sort(unique(as.vector(as.matrix(dat3$dass21_as[, dass21_as_mean_items])))))
+all(dass21_as_mean_range >= 0 & dass21_as_mean_range <= 3)
+
+# ---------------------------------------------------------------------------- #
+# Reverse-code anxiety identity ----
+# ---------------------------------------------------------------------------- #
+
+dat3$anxiety_identity$anxiety_identity_rev <-
+  abs(dat3$anxiety_identity$anxiety_identity - 4)
```

Comment:
The math is the same, but maybe more parsimonious to just say 4 - [x] rather than abs([x] - 4)

Reply:
Sounds great. Feel free to change!